src/hpr_routing.erl

```diff
@@ -97,95 +103,159 @@ find_routes({uplink, {_Type, DevAddr}}, PacketUp) ->
     ok = hpr_metrics:observe_find_routes(Time),
     Results.
 
+-spec start_crawl_routing() -> pid().
+start_crawl_routing() ->
+    erlang:spawn(fun() ->
+        Timer = timer:seconds(hpr_utils:get_env_int(routing_cleanup_timer_secs, 10)),
+        Window = timer:seconds(hpr_utils:get_env_int(routing_cleanup_window_secs, 60)),
+        ok = timer:sleep(Timer),
+        Removed = do_crawl_routing(Window),
+        lager:debug(
+            [{timer, Timer}, {window, Window}, {removed, Removed}],
+            "cleaning routing table"
+        ),
+        true = erlang:unregister(?ROUTING_CLEANUP),
```

Comment:
Why are you registering a name here? You can carry the process pid around in your gen server state and fire off a cast or info message to send messages to it.

Reply:
This module is not a gen_server.